apps/src/weblab2/Editor.tsx

```diff
@@ -0,0 +1,65 @@
+import React, {useCallback, useState} from 'react';
+
+import {useCDOIDEContext, editableFileType, prettify} from 'cdo-ide-poc';
+
+import CodeEditor from '@cdo/apps/lab2/views/components/editor/CodeEditor';
+import {html} from '@codemirror/lang-html';
+import {css} from '@codemirror/lang-css';
+import {javascript as js} from '@codemirror/lang-javascript';
+import {LanguageSupport} from '@codemirror/language';
+//import prettier from 'prettier/standalone';
+//import htmlParser from 'prettier/parser-html';
+//import cssParser from 'prettier/parser-postcss';
+
+const codeMirrorLangMapping: {[key: string]: LanguageSupport} = {
+  html: html(),
+  css: css(),
+  js: js(),
+};
+
+const Editor = () => {
+  const {project, saveFile} = useCDOIDEContext();
+  const file = Object.values(project.files).filter(f => f.active)?.[0];
+  // this is a stupid hack. the low level code-mirror editor won't update itself
+  // automatically if the doc is changed externally. So in lieu of doing it better,
+  // for now we just key the component off of the file ID + an incrementing value that
+  // hits every time the format button is pressed. That'll force a re-render and make it work.
+  // let's swap this out with something better.
+  const [formats, setFormats] = useState(1);
+
+  const onChange = useCallback(
+    (value: string) => {
+      saveFile(file.id, value);
```

Comment:
something to note is we will always save all files together

Reply:
All files are only always saved together. :-) 

That `saveFile` function there is internal to the editor, and what it's doing is is just updating the project json that was handed into the IDE.

_That_ updated json is then handed back to the `setProject` call back that was handed in when the IDE was instantiated, and it relies upon an external process to write that whole JSON blob back out to disk.